From 08ad05a39c060f281d2a0c555bf5ff535d88b88c Mon Sep 17 00:00:00 2001
From: Manish Verma <quic_maniverm@quicinc.com>
Date: Fri, 29 Mar 2024 18:05:05 +0530
Subject: [PATCH] net: Add new Kernel config for the latency measurement
 scenarios

Add new CONFIG_SKB_TIMESTAMP Kernel config which can be used to
enable the latency measurement scenarios. It adds additional storage
space in the packet descriptor for various timestamp values required
for the latency measurement

Change-Id: I75fa38bfe17a7772d6723279167b484bffaf3ae9
Signed-off-by: Manish Verma <quic_maniverm@quicinc.com>
---
 include/linux/skbuff.h | 8 ++++++++
 net/Kconfig            | 4 ++++
 2 files changed, 12 insertions(+)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index 6b06c0dbc3fc..a7eabac5354f 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -877,6 +877,14 @@ struct sk_buff {
 		ktime_t		tstamp;
 		u64		skb_mstamp_ns; /* earliest departure time */
 	};
+
+#ifdef CONFIG_SKB_TIMESTAMP
+	struct {
+		u64		delta_ts0;
+		u64		delta_ts1;
+	};
+#endif
+
 	/*
 	 * This is the control buffer. It is free to use for every
 	 * layer. Please put your private variables there. If you
diff --git a/net/Kconfig b/net/Kconfig
index d422dad56766..53754fd2fe9a 100644
--- a/net/Kconfig
+++ b/net/Kconfig
@@ -415,6 +415,10 @@ config ALLOC_SKB_PAGE_FRAG_DISABLE
 	help
 	 Disable page fragment based allocations for skbuff payloads.
 
+config SKB_TIMESTAMP
+	bool "Timestamp storage enablement in the packet"
+	default n
+
 menu "Network testing"
 
 config NET_PKTGEN
-- 
2.34.1

